control-plane-operator/controllers/hostedcontrolplane/cloud/azure/providerconfig.go

```diff
@@ -43,8 +44,19 @@ func ReconcileCloudConfigWithCredentials(secret *corev1.Secret, hcp *hyperv1.Hos
 		return err
 	}
 
-	cfg.AADClientID = hcp.Spec.Platform.Azure.ManagedIdentities.ControlPlane.CloudProvider.ClientID
-	cfg.AADClientCertPath = config.ManagedAzureCertificatePath + hcp.Spec.Platform.Azure.ManagedIdentities.ControlPlane.CloudProvider.CertificateName
+	// Get the ClientID and CertificateName config for csi drivers in ARO
+	switch secret.Name {
+	case manifests.AzureDiskConfigWithCredentials(hcp.Namespace).Name:
```

Comment:
I don't think this is the right fix for this issue. We are already reconciling the right info for the secrets here- https://github.com/openshift/hypershift/blob/74287710a908357c17902185ca2cf5f3f58886c9/control-plane-operator/controllers/hostedcontrolplane/storage/azure.go#L28-L27

Reply:
Looks like we just need to add the certificate path to the above^

Reply:
Those secrets are reconciled again and overriden here https://github.com/openshift/hypershift/blob/main/control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go#L5094-L5106